src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -148,190 +169,231 @@ def setUp(self):
         self.jc.stop_job_status_loop()
         self.job_states = get_test_job_states()
 
-    def check_error_message(self, req, source, input_, err):
+    def check_error_message(self, req, err, extra_params={}):
         """
         response when no input was submitted with a query
         args:
             req: the JobRequest, dictionary, or string passed when throwing the error
+            extra_params: any extra content
+            err: the error object produced
+
+        error message produced will contain:
+            raw_request: the input that generated the error
             source: the request that was called
-            input: the dict with job_id/job_id_list key and value
+            extra_params
             name: the type of exception
             message: the error message
         """
-        if isinstance(req, dict):
-            raw_request = req["content"]["data"]
-        elif isinstance(req, JobRequest):
+        if isinstance(req, JobRequest):
             raw_request = req.raw_request
+            source = req.request_type
+        elif isinstance(req, dict):
+            raw_request = req
+            source = req["content"]["data"]["request_type"]
         elif isinstance(req, str):
             raw_request = req
+            source = req
 
         msg = self.jc._comm.last_message
         self.assertEqual(
             {
                 "msg_type": ERROR,
                 "content": {
+                    "raw_request": raw_request,
                     "source": source,
-                    **input_,
+                    **extra_params,
                     "name": type(err).__name__,
                     "message": str(err),
-                    "raw_request": raw_request,
                 },
             },
-            msg["data"],
+            msg,
         )
 
     def check_job_id_list__no_jobs(self, request_type):
         job_id_list = [None, ""]
-        req = make_comm_msg(request_type, job_id_list, False)
-        err = JobIDException(JOBS_MISSING_FALSY_ERR, job_id_list)
+        req_dict = make_comm_msg(request_type, job_id_list, False)
+        req = make_comm_msg(request_type, job_id_list, True)
+        err = JobIDException(JOBS_MISSING_ERR, job_id_list)
+
+        # using handler
         with self.assertRaisesRegex(type(err), re.escape(str(err))):
-            self.jc._handle_comm_message(req)
-        self.check_error_message(req, request_type, {JOB_ID_LIST: job_id_list}, err)
+            self.jc._handle_comm_message(req_dict)
+        self.check_error_message(req_dict, err)
```

Comment:
check both the `_handle_comm_message` version and the `_msg_map` way of invoking a method

Reply:
changed the tests to test via `_handle_comm_message` generally since it wraps nearly all the functions performed by the jobcomm.